spec/controllers/applications_controller_spec.rb

```diff
@@ -15,6 +15,21 @@
         get :index
         expect(assigns[:rss]).to be_nil
       end
+
+      it "should try to convert page param to an integer" do
```

Comment:
In my first review I didn't check where these tests were placed. They're under `#index rss feed` and since this isn't related to the RSS feed these tests should not be here.

But where should they go? I didn't know so I hunted around. I discovered that `ApplicationController` behaviour that does things like global error checking should be tested by [defining an anonymous controller](https://www.relishapp.com/rspec/rspec-rails/v/3-0/docs/controller-specs/anonymous-controller).

So create a new file in this directory called `application_controller_spec.rb`, with something like this (and removing the tests from this file):

```ruby
require "spec_helper"

describe ApplicationController do
  controller do
    def index
      render text: nil
    end
  end

  describe "#validate_page_param" do
    it "should try to convert page param to an integer" do
      get :index, page: "2%5B&q"

      expect(controller.params[:page]).to eq(2)
    end

    it "should default to page nil when no page number param is given" do
      get :index, page: "%5B&q"

      expect(controller.params[:page]).to eq(nil)
    end
  end
end
```